src/modules/user/user.service.ts

```diff
@@ -44,6 +46,10 @@ export default class UserService {
     return await GetRecord[identifierType]();
   }
 
+  async saveUser(user: Partial<UserInterface>) {
```

Comment:
I actually have some reservations on this method. Does this method not create a new user ? If yes then there is an existing method for that in the user service

Reply:
It doesn't create a new user. It just calls the save method on the user that I retrieved in the 2fa method. The verify2fa method needs to update the two_factor secret and backup_codes records on the user's table.
An alternative method which I did not have at the time of writing the code is the pass the updates into the updateUser method. @Homoakin619

Reply:
I've removed that method and made use of the updateUserRecord